provider-service/base/pkg/server/server.go

```diff
@@ -144,10 +144,19 @@ func newHandler(resources []resource.HttpHandledResource) http.Handler {
 	return mux
 }
 
-func Start(ctx context.Context, cfg config.Server, provider resource.Provider) error {
-	addr := fmt.Sprintf("%s:%d", cfg.Host, cfg.Port)
+func Start(ctx context.Context, cfg config.Config, provider resource.Provider) error {
+	addr := fmt.Sprintf("%s:%d", cfg.Server.Host, cfg.Server.Port)
 	logger := common.LoggerFromContext(ctx)
 
+	onShutdown, err := InitialiseMetricsServer(ctx, cfg.Metrics)
```

Comment:
I don't think this should be initialised here, it should be within main or there should be a new structure
```
 | server
    |- provider_server.go
    | # contains the functionality to spin up the provider http server
    |- metrics_server.go
    | # contains the functionalitu to spin up the metrics server
    |_ server.go
      # helper to spin start both servers and used within main.go

```